UI/window-basic-main-profiles.cpp

```diff
@@ -783,6 +783,16 @@ void OBSBasic::ChangeProfile()
 	auth.reset();
 	DestroyPanelCookieManager();
 
+#ifdef YOUTUBE_ENABLED
+	OBSBasic *main = OBSBasic::Get();
+	// This dock works with both stream key and auth
+	// so hide it here if required.
+	YouTubeAppDock *youtubeAppDock = main->GetYouTubeAppDock();
+	if (youtubeAppDock != NULL) {
+		youtubeAppDock->hide();
+	}
```

Comment:
Since this method is in `OBSBasic` you can just access the app dock variable directly.
```suggestion
	if (youtubeAppDock)
		youtubeAppDock->hide();
```